File: showsclipboard/src/main/java/com/csr1/showsclipboard/MainActivity.java
package com.csr1.showsclipboard;

import androidx.appcompat.app.AppCompatActivity;

import android.app.ActivityManager;
import android.content.ClipboardManager;
import android.content.Context;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    ClipboardManager clipboardManager;
    TextView textView;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        textView = findViewById(R.id.tv_clip);
        clipboardManager = (ClipboardManager) getSystemService(Context.CLIPBOARD_SERVICE);
        clipboardManager.addPrimaryClipChangedListener(new ClipboardManager.OnPrimaryClipChangedListener() {
            @Override
            public void onPrimaryClipChanged() {
                String s = clipboardManager.getPrimaryClip().getItemAt(0).coerceToHtmlText(MainActivity.this);
                textView.append("\n"+s);
            }
        });
    }

    public void onClick(View view) {
        String s = clipboardManager.getPrimaryClip().getItemAt(0).coerceToText(MainActivity.this).toString();
        textView.append("\n"+s);
    }

}

File: somecrypto/src/main/java/com/hackeru/somecrypto/MainActivity.kt
package com.hackeru.somecrypto

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.security.keystore.KeyInfo
import android.view.TextureView
import android.view.View
import android.widget.TextView
import java.security.KeyStore
import javax.crypto.SecretKey
import javax.crypto.SecretKeyFactory

class MainActivity : AppCompatActivity() {
    lateinit var tv:TextView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        tv = findViewById(R.id.textView)
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) {
            val keyStore = KeyStore.getInstance("AndroidKeyStore")
            keyStore.load(null,null)
            val key = keyStore.getKey("key1","abcdefg".toCharArray())
//            val keyInfo = SecretKeyFactory.getInstance(key.algorithm).getKeySpec(key as SecretKey?,KeyInfo::class.java)
            tv.text = "type: " + keyStore.type+"\n"
            tv.append("provider: "+keyStore.provider+"\n")
            tv.append("aliases: "+keyStore.aliases().toList()+"\n")
            tv.append("size: "+keyStore.size()+"\n")
            tv.append("str: "+keyStore.toString()+"\n")
            tv.append("-----------------------------------------------------------------\n")
//            val mainKey = MasterKey.Builder(applicationContext)
//                    .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
//                    .build()


        }
    }

    fun onClick(view: View) {

    }
}